packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -0,0 +1,316 @@
+  display?: ResultDisplayLayout;
+}
+export interface ResultPlaceholderProps extends Omit<DisplayOptions, 'image'> {
+  resultsPerPageState: ResultsPerPageState;
+}
+export interface ResultsProps extends DisplayOptions {
+  resultListState: FoldedResultListState | ResultListState;
+  resultListCommon: ResultListCommon;
+  bindings: Bindings;
+  host: HTMLElement;
+  getContentOfResultTemplate(
+    result: Result | FoldedResult
+  ): HTMLElement | DocumentFragment;
+
+  classes: string;
+}
+
+interface TemplateElement extends HTMLElement {
+  getTemplate(): Promise<ResultTemplate<DocumentFragment> | null>;
+}
+
+interface ResultListCommonOptions {
+  bindings: Bindings;
+  templateElements: NodeListOf<TemplateElement>;
+  fieldsToInclude?: string;
+  includeDefaultTemplate?: boolean;
+  onReady(): void;
+  onError(): void;
+}
+
+export class ResultListCommon {
+  private bindings: Bindings;
+  private render?: ResultRenderingFunction;
+  private updateBreakpoints?: (host: HTMLElement) => void;
+
+  public resultTemplatesManager!: ResultTemplatesManager<TemplateContent>;
+  public resultListControllerProps?: ResultListProps;
+
+  constructor(opts: ResultListCommonOptions) {
+    this.bindings = opts.bindings;
+    this.updateBreakpoints = once((host: HTMLElement) => {
+      updateBreakpoints(host);
+    });
+
+    if (opts.fieldsToInclude) {
+      this.resultListControllerProps = {
+        options: {
+          fieldsToInclude: this.determineAllFieldsToInclude(
+            opts.fieldsToInclude
+          ),
+        },
+      };
+    }
+
+    this.registerResultTemplates(
+      opts.templateElements,
+      opts.includeDefaultTemplate,
+      opts.onReady,
+      opts.onError
+    );
+  }
+
+  set renderingFunction(render: ResultRenderingFunction) {
+    this.render = render;
+  }
+
+  private determineAllFieldsToInclude(
+    configuredFieldsToInclude: string
+  ): string[] {
+    if (configuredFieldsToInclude.trim() === '')
+      return [...EcommerceDefaultFieldsToInclude];
+    return EcommerceDefaultFieldsToInclude.concat(
+      configuredFieldsToInclude.split(',').map((field) => field.trim())
+    );
+  }
+
+  public getTemplate(result: Result) {
+    return this.resultTemplatesManager.selectTemplate(result);
+  }
+
+  public getContentOfResultTemplate(
+    resultOrFolded: Result | FoldedResult
+  ): HTMLElement | DocumentFragment {
+    const result = (resultOrFolded as FoldedResult).result || resultOrFolded;
+    return this.render ? this.render(result) : this.getTemplate(result)!;
+  }
+
+  private makeDefaultTemplate(): ResultTemplate<DocumentFragment> {
+    const content = document.createDocumentFragment();
+    const linkEl = document.createElement('atomic-result-link');
+    content.appendChild(linkEl);
+    return {
+      content,
+      conditions: [],
+    };
+  }
+
+  public async registerResultTemplates(
+    elements: NodeListOf<TemplateElement>,
+    includeDefaultTemplate = true,
+    onReady: () => void,
```

Comment:
Instead of taking `onReady` as a param, the caller could just `await` this function.

Instead of taking `onError` as a param, perhaps we could `throw` here and `catch` from the caller.

Reply:
the `onReady` function has the problem of initialize not being awaited either. I'm looking into it in a separate task, but it's looking very, very tricky. I'll leave it as is for now.
about the `onError`, I personally think it looks neater like this instead of then implementing the catch on each component. having it as pros makes it more obvious, I think!
I'll merge like this anyway and can always change after!